src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -327,6 +328,11 @@ mongoc_topology_new (const mongoc_uri_t *uri, bool single_threaded)
    if (service) {
       memset (&rr_data, 0, sizeof (mongoc_rr_data_t));
 
+      /* Upon failure, the SRV polling thread still needs a duration. */
```

Comment:
What does "upon failure" mean? Is this to ensure you set these in case a jump is made to `srv_fail` later on?

Do I assume correctly that this has no impact on the original assignments later on and those should be left as-is?